kvdb/postgres/readwrite_tx.go

```diff
@@ -39,7 +40,16 @@ func newReadWriteTx(db *db, readOnly bool) (*readWriteTx, error) {
 	}
 	locker.Lock()
 
-	tx, err := db.db.Begin()
+	// Start the transaction. Don't use the timeout context because it would
+	// be applied to the transaction as a whole. If possible, mark the
+	// transaction as read-only to make sure that potential programming
+	// errors cannot cause changes to the database.
+	tx, err := db.db.BeginTx(
```

Comment:
Is this actually what we want given this function is called `newReadWriteTx` and returns a rw transaction?

I would think instead we'd just be using the read transaction to load these items.